ludwig/automl/base_config.py

```diff
@@ -222,20 +222,23 @@ def get_dataset_info(df: Union[pd.DataFrame, dd.core.DataFrame]) -> DatasetInfo:
 
 
 def is_field_boolean(source: DataSource, field: str) -> bool:
-    num_unique_values, unique_values, _ = source.get_distinct_values(field, max_values_to_return=4)
+    """Returns a boolean indicating whether the object field should have a bool dtype.
+
+    Columns with object dtype that have 3 distinct values of which one is Nan/None is a bool type column.
+    """
+    unique_values = source.df[field].unique()
+    num_unique_values = len(unique_values)
     if num_unique_values <= 3:
-        for entry in unique_values:
+        if None in unique_values:
```

Comment:
Why does this mean that it is automatically True?

Reply:
So we know the field has an object dtype, and we're trying to see if we should manually set the type to bool. Bool types have 2 distinct values, but if there is a None or a NaN, then it had 3 distinct values which is why it's being inferred as an object. I added the None conditional check incase the field is just empty and not an np.nan, but I can also just remove this check.

Reply:
what if unique values are [NaN, "Bananas", None]? Wouldn't we then see the nan, return True from the function even though it isn't true?

Reply:
Hmm, that is an interesting edge case. Would the dtype in this case just be object?

Reply:
>The bug is valid but I thought the only case where we want to change from type object to bool is if we have two bool values and a NaN

That makes sense, will update

Reply:
@magdyksaleh just updated